client/state/admin-menu/schema.js

```diff
@@ -0,0 +1,37 @@
+const commonItemPropsSchema = {
+	icon: { type: 'string' },
+	slug: { type: 'string' },
+	title: { type: 'string' },
+	type: { type: 'string' },
+	url: { type: 'string' },
+};
+
+const menuItemsSite = {
+	type: 'array',
+	items: {
+		type: 'object',
+		required: [ 'type' ],
+		properties: {
+			...commonItemPropsSchema,
+			children: {
+				type: 'array',
+				items: {
+					type: 'object',
+					required: [ 'type', 'parent' ],
+					properties: {
+						...commonItemPropsSchema,
```

Comment:
children will not have an `icon`. It might be clearer to just keep the schema for parents and children inline

Reply:
Nice spot. Thank you!

If we're using browser persistence I'd rather that we're capturing the props accurately to ensure any changes to the API schema are managed correctly. Failure to do this could lead to errors for our users.

Therefore I've removed `icon` from the common props and made it on the `parent` items only.